lib/Sema/CSRanking.cpp

```diff
@@ -977,7 +977,36 @@ ConstraintSystem::compareSolutions(ConstraintSystem &cs,
       ++score1;
     else if (isa<VarDecl>(decl2) && isa<FuncDecl>(decl1))
       ++score2;
-    
+
+    // If both are class properties with the same name/type, prefer
+    // the one wich is attached to the subclass because it could be
+    // found only if requested directly.
+    if (!decl1->isInstanceMember() && !decl2->isInstanceMember()) {
+      auto type1 = decl1->getInterfaceType();
+      auto type2 = decl2->getInterfaceType();
+
+      if (isa<VarDecl>(decl1) && isa<VarDecl>(decl2) && type1->isEqual(type2)) {
+        auto *nominal1 = dc1->getAsNominalTypeOrNominalTypeExtensionContext();
+        auto *nominal2 = dc2->getAsNominalTypeOrNominalTypeExtensionContext();
+
+        if (nominal1 && nominal2 && nominal1 != nominal2) {
+          auto baseType = choice1.getBaseType();
+          if (auto MTT = baseType->getAs<AnyMetatypeType>()) {
+            auto instanceType = MTT->getInstanceType();
+
+            auto base1 = nominal1->getDeclaredType();
+            auto base2 = nominal2->getDeclaredType();
+
+            if (base1->isEqual(instanceType))
```

Comment:
I originally tried to do that, check if base1 is superclass of base2, but it seems to me if we are requesting something on the subclass and trying to prioritize that, we should always check if the base we have requested equals to the base of the result and take that because I am also not sure how to check "closer in the inheritance chain"...